crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+        };
+        Ok(response)
+    }
+
+    fn map_failed_transaction_preview(
+        response: TransactionPreviewResponse,
+    ) -> CommonError {
+        let message = response
+            .receipt
+            .error_message
+            .unwrap_or_else(|| "Unknown reason".to_string());
+
+        // Quite rudimentary, but it is not worth making something smarter,
+        // as the GW will provide in the future strongly typed errors
+        let is_failure_due_to_deposit_rules = message
+            .contains("AccountError(DepositIsDisallowed")
+            || message.contains("AccountError(NotAllBucketsCouldBeDeposited");
+
+        if is_failure_due_to_deposit_rules {
+            CommonError::OneOfReceivingAccountsDoesNotAllowDeposits
+        } else {
+            CommonError::FailedTransactionPreview {
+                error_message: message,
+            }
+        }
+    }
+
+    async fn extract_transaction_signers(
+        &self,
+        manifest_summary: ManifestSummary,
+    ) -> Result<IndexSet<PublicKey>> {
+        let signer_entities = self
+            .extract_transaction_signer_entities(manifest_summary.clone())
+            .await?;
+        Ok(self.extract_transaction_signers_public_keys(signer_entities))
+    }
+
+    async fn extract_transaction_signer_entities(
+        &self,
+        manifest_summary: ManifestSummary,
+    ) -> Result<IndexSet<AccountOrPersona>> {
+        let accounts =
+            self.profile_state_holder.accounts_on_current_network()?;
+        let personas =
+            self.profile_state_holder.personas_on_current_network()?;
+
+        let account_entities: IndexSet<_> = manifest_summary
+            .addresses_of_accounts_requiring_auth
+            .iter()
+            .filter_map(|address| {
+                accounts
+                    .iter()
+                    .find(|account| account.address == *address)
+                    .map(AccountOrPersona::AccountEntity)
+            })
+            .collect();
+        let persona_entities: IndexSet<_> = manifest_summary
+            .addresses_of_personas_requiring_auth
+            .iter()
+            .filter_map(|address| {
+                personas
+                    .iter()
+                    .find(|persona| persona.address == *address)
+                    .map(AccountOrPersona::PersonaEntity)
+            })
+            .collect();
+        let mut signer_entities: IndexSet<AccountOrPersona> = IndexSet::new();
+        signer_entities.extend(account_entities);
+        signer_entities.extend(persona_entities);
+        Ok(signer_entities)
+    }
+
+    fn extract_transaction_signers_public_keys(
+        &self,
+        signers: IndexSet<AccountOrPersona>,
+    ) -> IndexSet<PublicKey> {
+        signers
+            .iter()
+            .flat_map(|entity| {
+                let public_keys: IndexSet<PublicKey> = entity
+                    .virtual_hierarchical_deterministic_factor_instances()
```

Comment:
ah, you want to use a different function for the signeres, since the current impl of : `virtual_hierarchical_deterministic_factor_instances` includes ROLA keys, with `CAP26KeyKind::AuthenticationSigning` which cannot be used for `TransctionSigning`.

And since this method is called "transaction_signers"` you wanna FILTER to only include the public keys of kind TransactionSigning.

Furthermore this method should probably return `HiearchalDeterministicFactorInstance` instead of `PublicKey`, or rather you should create a `extract_transaction_signers_factor_instances(&self, signers: IndexSet<AccountOrPersona>) -> IndexSet<HiearchalDeterministicFactorInstance>` 

and then the `extract_transaction_signers_public_keys` can simply call `extract_transaction_signers_factor_instances` peforming the `map(|fi| fi.public_key.public_key)`..